src/github.com/matrix-org/dendrite/roomserver/api/query.go

```diff
@@ -68,6 +68,22 @@ type QueryStateAfterEventsResponse struct {
 	StateEvents []gomatrixserverlib.Event
 }
 
+// QueryEventsByIDRequest is a request to QueryEventsByID
+type QueryEventsByIDRequest struct {
+	// The event IDs to look up.
+	EventIDs []string
+}
+
+// QueryEventsByIDResponse is a response to QueryEventsByID
+type QueryEventsByIDResponse struct {
+	// Copy of the request for debugging.
+	QueryEventsByIDRequest
+	// A list of events with the request IDs.
+	// If the roomserver does not have a copy of a requested event
+	// then it will omit that event from the list.
+	Events []gomatrixserverlib.Event
```

Comment:
Probably also worth explicitly stating that if the roomserver HAS a copy of the event but is unable to load it, it will return an error rather than being omitted.